src/Microsoft.OData.Edm/Csdl/Parsing/SchemaJsonParser.cs

```diff
@@ -0,0 +1,1564 @@
+                        break;
+
+                    case "$UnderlyingType":
+                        // The value of $UnderlyingType is the qualified name of the underlying type.
+                        underlyingTypeName = propertyValue.ParseAsString(context);
+                        break;
+
+                    case "$IsFlags":
+                        // The value of $IsFlags is one of the Boolean literals true or false.Absence of the member means false.
+                        isFlags = propertyValue.ParseAsBoolean(context);
+                        break;
+
+                    default:
+                        string termName;
+                        propertyName = SeperateAnnotationName(propertyName, out termName);
+                        if (termName == null)
+                        {
+                            // Without '@' in the name, so it's normal enum member name
+                            CsdlEnumMember enumMember = ParseCsdlEnumMember(propertyName, propertyValue, context);
+                            members.Add(enumMember);
+                            break;
+                        }
+                        else
+                        {
+                            if (AnnotationJsonParser.TryParseCsdlAnnotation(termName, propertyValue, context, out CsdlAnnotation annotation))
+                            {
+                                if (propertyName == null)
+                                {
+                                    // annotation for the enum type
+                                    enumAnnotations.Add(annotation);
+                                }
+                                else
+                                {
+                                    // annotation for enum member
+                                    IList<CsdlAnnotation> values;
+                                    if (!memberAnnotations.TryGetValue(propertyName, out values))
+                                    {
+                                        values = new List<CsdlAnnotation>();
+                                        memberAnnotations[propertyName] = values;
+                                    }
+                                    values.Add(annotation);
+                                }
+
+                                break;
+                            }
+                        }
+
+                        ReportUnexpectedElement(propertyValue, context);
+                        break;
+                }
+            });
+
+            foreach (var item in memberAnnotations)
+            {
+                // TODO: maybe refactor later for performance
+                CsdlEnumMember member = members.FirstOrDefault(c => c.Name == item.Key);
+                if (member == null)
+                {
+                    context.ReportError(EdmErrorCode.JsonMissingRequiredProperty, "Cannot find enum member that an annotation applys for.");

```

Comment:
>ReportError [](start = 28, length = 11)

this should be a non-fatal error

Reply:
change to "UnexpectedJsonElement"

---
In reply to: [443649594](https://github.com/OData/odata.net/pull/1775#discussion_r443649594) [](ancestors = 443649594)